app/test_engine/models/manual_test_case.py

```diff
@@ -153,7 +153,10 @@ def handle_uploaded_file(self, file: UploadFile) -> None:
         logger.info("---- Start of Manual Log ----")
         with file.file as f:
             for line in f:
-                logger.info(line.decode("utf-8").strip())
+                if file.content_type == "application/octet-stream":
+                    logger.info(line.decode("utf-8").strip())
+                else:
+                    logger.info(line.strip())
```

Comment:
As discussed offline `line` here is always bytes, and we should always `decode` it as `utf-8`. This is not related to the `content_type` of the uploaded log.

The real problem here is that some `line` could contain invalid UTF-8 sequences. 

We have a couple options on how to handle it:
1. Simply reject any file that contains non-UTF-8 characters. 
2. Gracefully attempt to handle the places that this occur.

This proposal is one way to do it gracefully.

```suggestion
                try:
                    logger.info(line.decode("utf-8").strip())
                except UnicodeDecodeError:
                    logger.warning("WARNING: The following line contained invalid UTF-8. Some content was replaced with: �")
                    logger.info(line.decode("utf-8", errors='replace').strip())
```

Reply:
I agreed